plugin-server/src/main/ingestion-queues/session-recording/session-recordings-consumer-v3.ts

```diff
@@ -0,0 +1,450 @@
+            logExecutionTime: true,
+            func: async () => {
+                histogramKafkaBatchSize.observe(messages.length)
+                histogramKafkaBatchSizeKb.observe(messages.reduce((acc, m) => (m.value?.length ?? 0) + acc, 0) / 1024)
+
+                const recordingMessages: IncomingRecordingMessage[] = []
+
+                await runInstrumentedFunction({
+                    statsKey: `recordingingester.handleEachBatch.parseKafkaMessages`,
+                    func: async () => {
+                        for (const message of messages) {
+                            counterKafkaMessageReceived.inc({ partition: message.partition })
+
+                            const recordingMessage = await parseKafkaMessage(message, (token) =>
+                                this.teamsRefresher.get().then((teams) => ({
+                                    teamId: teams[token]?.teamId || null,
+                                    consoleLogIngestionEnabled: teams[token]?.consoleLogIngestionEnabled ?? true,
+                                }))
+                            )
+
+                            if (recordingMessage) {
+                                recordingMessages.push(recordingMessage)
+                            }
+                        }
+                    },
+                })
+
+                // await this.reportPartitionMetrics()
+
+                await runInstrumentedFunction({
+                    statsKey: `recordingingester.handleEachBatch.ensureSessionsAreLoaded`,
+                    func: async () => {
+                        await this.syncSessionsWithDisk()
+                    },
+                })
+
+                await runInstrumentedFunction({
+                    statsKey: `recordingingester.handleEachBatch.consumeBatch`,
+                    func: async () => {
+                        for (const message of recordingMessages) {
+                            await this.consume(message)
+                        }
+                    },
+                })
+
+                await runInstrumentedFunction({
+                    statsKey: `recordingingester.handleEachBatch.flushAllReadySessions`,
+                    func: async () => {
+                        await this.flushAllReadySessions()
+                    },
+                })
+
+                // await runInstrumentedFunction({
+                //     statsKey: `recordingingester.handleEachBatch.consumeReplayEvents`,
+                //     func: async () => {
+                //         await this.replayEventsIngester.consumeBatch(recordingMessages)
+                //     },
+                // })
+
+                // await runInstrumentedFunction({
+                //     statsKey: `recordingingester.handleEachBatch.consumeConsoleLogEvents`,
+                //     func: async () => {
+                //         await this.consoleLogsIngester.consumeBatch(recordingMessages)
+                //     },
+                // })
```

Comment:
I'm guessing because we're just going to test this consumer for perf and not have it write to CH yet